File: create-userDaily-Calories - ServerSide/src/main/java/in/satckroute/model/Dieter.java
package in.satckroute.model;

import in.satckroute.model.MacroNutrient;
import jakarta.persistence.*;
import lombok.*;
//import lombok.AllArgsConstructor;
//import lombok.Getter;
//import lombok.NoArgsConstructor;
//import lombok.Setter;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Dieter {

    @Id
    @GeneratedValue
    private int dieterId;

    @Column(nullable = false)
    private String gender;

    @Column(nullable = false)
    private int age;

    @Column(nullable = false)
    private double weight;

    @Column(nullable = false)
    private double height;

    @Column(nullable = false)
    private String activityLevel;

    @Column(nullable = false)
    private String weightGoal;

    @Column(nullable = false)
    private int mainMeals_No;

    @Column(nullable = false)
    private int snacks_No;

    @Column(nullable = false)
    private float dailyCalories;


//    @OneToOne(cascade = CascadeType.ALL)
//    @JoinColumn(name = "macro_nutrient_id")
//    private MacroNutrient macroNutrient;
}

File: product-webapp-service/webapp/src/app/model/dieter.ts
export class Dieter {
    dieterId: number | null = null;
    gender: string = '';
    age: number | null = null;
    weight: number | null = null;
    height: number | null = null;
    activityLevel: string = '';
    weightGoal: string = '';
    mainMeals_No: number | null = null;
    snacks_No: number | null = null;
    dailyCalories: number | null = null;

    constructor(
        dieterId: number | null = null,
        gender: string = '',
        age: number | null = null,
        height: number | null = null,
        weight: number | null = null,
        activityLevel: string = '',
        weightGoal: string = '',
        mainMeals_No: number | null = null,
        snacks_No: number | null = null,
        dailyCalories: number | null=null
    ) {
        // this.dieterId = dieterId;
        this.gender = gender;
        this.age = age;
        this.height = height;
        this.weight = weight;
        this.activityLevel = activityLevel;
        this.weightGoal = weightGoal;
        this.mainMeals_No = mainMeals_No;
        this.snacks_No = snacks_No;
        this.dailyCalories= dailyCalories;
    }
}
